User details update

burakkarasel opened the pull request:
Updated Role Enum for the needs. 
Changed User Entity to store only auth related fields. 
Changed Entity ID's from Integer to UUID. 
Added new Gender Enum.
Added new UserOtherDetails Entity to store non-auth related fields.
Created a OneToOne relation between User and UserOtherDetails.
Created new Service and Repository Implementations for UserOtherDetails.
Updated register functionality for the new changes.
Added new dependency for input validation.
Added annotations to request dtos.
Added custom exception handler for input validation.
Added new dependency ObjectMapper to convert test bodies to json.
Added new unit tests for register and authenticate functionality.
Updated regex pattern for email and password validation.
Added new AuthenticationExceptionHandler.
Added new UserService.
Added new InsuranceInformationService.
Added new RoleChangeNotPossibleException and it's handler.
Added new CreatePatientRequest and CreateResponse dtos.
Implemented Create a new Patient functionality.
Added new SpringSecurityUserProvider test configuration to mock authenticated user.
Changed AuthenticationController's location from service to controller package.
Added new PatientControllerTest to test creating a new patient functionality.
Changed Role enum, added PatientDto.
Added new repository and service method to find patient by user.
Added controller for patient details with pre authorization for patient role.
Added new test cases for the new service

mustafaboleken commented:
Other than my notes, I will extend the unit tests. Then, all of us can use the same aproach.

mustafaboleken commented:
We need to add logs

mustafaboleken commented:
I just integrate the open api 3, you can pull from main and add API descriptions

mustafaboleken commented:
Existing tests are not passing

burakkarasel commented:
Added new NoSuchElementExceptionHandler.
Updated patient create service to accept UserDetails not User entity.
Updated tests accordingly.
Corrected few misspellings

burakkarasel commented:
Added new unique restriction to User entity.
Added new DataIntegrityViolationExceptionHandler to handle the exceptions.
Added new test case for register with duplicate email.

mustafaboleken reviewed (approved):
lgtm

burakkarasel commented on docker-compose.yml:
```diff
@@ -6,6 +6,7 @@ services:
     environment:
       - POSTGRES_USER=postgres
       - POSTGRES_PASSWORD=postgrespw
+      - POSTGRES_DATABASE=heu_plus
```
Not really😄

burakkarasel commented on src/main/java/com/bola/boilerplate/service/concretes/PatientService.java:
```diff
@@ -0,0 +1,71 @@
+
+    @Override
+    public PatientDto details(User user) {
+        Patient patient = repository.findByUser(user);
```
In the updated version I've used orElseThrow() which throws NoSuchElementException and it's handled by it's custom exception handler

burakkarasel commented on src/main/java/com/bola/boilerplate/service/concretes/InsuranceInformationService.java:
```diff
@@ -0,0 +1,23 @@
+                .policyNumber(policyNumber)
+                .build();
+
+        return repository.save(toCreate);
```
Handling now #6
